jupyter_client/manager.py

```diff
@@ -500,21 +554,23 @@ def restart_kernel(self, now=False, newports=False, **kw):
                                "No previous call to 'start_kernel'.")
         else:
             # Stop currently running kernel.
-            self.shutdown_kernel(now=now, restart=True)
+            await self._async_shutdown_kernel(now=now, restart=True)
```

Comment:
This breaks subclasses that override shutdown_kernel() as they won't have their override called during restarts.